Scenes/Entities/Entity.gd

```diff
@@ -0,0 +1,32 @@
+extends Node2D
+class_name Entity
+
+# --- Exported Properties ---
+@export var invincible: bool = false       # If true, the entity takes no damage.
+@export var max_health: int = 100          # Maximum health for the entity.
+
+# --- Public Properties ---
+static var map: TileMapLayer               # Shared reference to a TileMap layer.
+
+# --- Private Properties ---
+var _health: int = max_health              # Current health, initially set to max_health.
+@onready var map_pos: Vector2i = get_parent().local_to_map(position) # Position of the entity on the map.
+
+# --- Enum ---
+
+
+# --- Signals ---
+# Define custom signals here if needed, e.g., for health changes or status updates.
+# signal health_changed(new_health)
+
+# --- Built-in Callbacks ---
+func _ready() -> void:
+	# Called when the node enters the scene tree for the first time.
+	# Assign the map to the entity's parent, assuming it's a TileMap layer.
+	map = get_parent()
```

Comment:
There are some situations where an entity might not be a child of a the tilemap. Also generally things shouldn't be children of the map but that's beyond the scope of this PR. Recommend making a group or a global autoload to keep track of the map.

Reply:
> There are some situations where an entity might not be a child of a the tilemap. Also generally things shouldn't be children of the map but that's beyond the scope of this PR. Recommend making a group or a global autoload to keep track of the map.

Okay, so a couple questions for my understanding. 1. What is wrong with entities being children of the tile map? Is there an offset issue or something else? 2. Can you give me a scenario where an entity would not be part of the tile map, so i know what to prepare for? (i tried to make the entire system easily configureable via tilemap, becuse everything is going to be procedurly generated.)

Reply:
I find that when I have a lot of nodes in a scene I often group them under other nodes to make hiding them and filtering them easier, for example I've got a node that holds all the players, all the ship systems, all the props, etc that way I can mass operate on them without much boilerplate. It also means we can move the map around if we for some reason want to do that. If we're making the player a scene tile then that should be ok, not sure if I agree w/that implementation but idk if that's what we're doing tbh.

Reply:
> I find that when I have a lot of nodes in a scene I often group them under other nodes to make hiding them and filtering them easier, for example I've got a node that holds all the players, all the ship systems, all the props, etc that way I can mass operate on them without much boilerplate. It also means we can move the map around if we for some reason want to do that. If we're making the player a scene tile then that should be ok, not sure if I agree w/that implementation but idk if that's what we're doing tbh.

yea, sorry i just have all ways wanted to implement the player as a tile, so when i got the chance i went with it.